frontend/public/components/debug-terminal.tsx

```diff
@@ -0,0 +1,226 @@
+  const infoMessage = (
+    <div className="padding-info-message">
+      <Alert
+        variant="info"
+        isInline
+        title={t(
+          'public~This temporary pod has a modified entrypoint command to debug a failing container. The pod will be available for one hour and will be deleted when the terminal window is closed.',
+        )}
+      />
+    </div>
+  );
+  switch (obj?.data?.status?.phase) {
+    case 'Failed':
+      return (
+        <DebugTerminalError
+          error={
+            <>
+              {t('public~The debug pod failed. ')}
+              {obj?.data?.status?.containerStatuses?.[0]?.state?.terminated?.message ||
+                obj?.data?.status?.message}
+            </>
+          }
+        />
+      );
+    case 'Running':
+      return (
+        <PodExecLoader obj={obj.data} activeContainer={activeContainer} infoMessage={infoMessage} />
+      );
+    default:
+      return <LoadingBox />;
+  }
+};
+
+export const DebugTerminal: React.FC<DebugTerminalProps> = (props) => {
+  const [resources, setResources] = React.useState<FirehoseResource[]>([]);
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const podToDebug = props.obj;
+  const podName = podToDebug?.metadata?.name;
+  const podNamespace = podToDebug?.metadata?.namespace;
+  const containerName = props.match.params.name
+    ? props.match.params.name
+    : props.obj?.spec?.containers?.[0].name;
+  React.useEffect(() => {
+    const debugPodName = `${podName}-${containerName}-debug`;
+    const deleteDebugPod = async (name) => {
+      try {
+        await k8sKillByName(PodModel, name, podNamespace);
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn('Could not delete container terminal debug pod.', e);
+      }
+    };
+    const createDebugPod = async () => {
+      try {
+        const podToCreate = await getDebugPod(debugPodName, podToDebug, containerName);
+        const debugPod = await k8sCreate(PodModel, podToCreate);
+        if (debugPod) {
+          setResources([
+            {
+              isList: false,
+              kind: 'Pod',
+              name: debugPodName,
+              namespace: podNamespace,
+              prop: 'obj',
+            },
+          ]);
+        }
+      } catch (e) {
+        setErrorMessage(e.message);
+      }
+    };
+    createDebugPod();
+    window.addEventListener('beforeunload', deleteDebugPod);
+    return () => {
+      deleteDebugPod(debugPodName);
+      window.removeEventListener('beforeunload', deleteDebugPod);
+    };
+  }, [containerName, podName, podNamespace, podToDebug]);
+
+  return errorMessage ? (
+    <DebugTerminalError error={errorMessage} />
+  ) : (
+    <Firehose resources={resources}>
+      <DebugTerminalInner activeContainer={containerName} />
+    </Firehose>
+  );
+};
+
+export const DebugTerminalPage: React.FC<DebugTerminalPageProps> = (props) => {
+  const { t } = useTranslation();
+  return (
```

Comment:
Futher destructure `params` to avoid duplication.

```suggestion
  const { match: { params: { podName } }, match: { params: { ns } }, match: { params: { name } }, match : { url } } = match;
  return ( 
```